lib/rules/no-empty-pattern.js

```diff
@@ -19,19 +21,52 @@ module.exports = {
             url: "https://eslint.org/docs/latest/rules/no-empty-pattern"
         },
 
-        schema: [],
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    allowObjectPatternsAsParameters: {
+                        type: "boolean",
+                        default: false
+                    }
+                },
+                additionalProperties: false
+            }
+        ],
 
         messages: {
             unexpected: "Unexpected empty {{type}} pattern."
         }
     },
 
     create(context) {
+        const options = context.options[0] || {},
+            allowObjectPatternsAsParameters = options.allowObjectPatternsAsParameters || false;
+
         return {
             ObjectPattern(node) {
-                if (node.properties.length === 0) {
-                    context.report({ node, messageId: "unexpected", data: { type: "object" } });
+
+                if (node.properties.length > 0) {
+                    return;
                 }
+
+                // Allow {} and {} = {} empty object patterns in Arrow functions as parameters when allowObjectPatternsAsParameters is true
```

Comment:
```suggestion
                // Allow {} and {} = {} empty object patterns as parameters when allowObjectPatternsAsParameters is true
```

This applies to all function definitions.